data.md

```diff
@@ -533,14 +536,14 @@ The function interprets the range of bytes as little-endian.
 `#set` sets a key in a map, removing the key if the value is 0.
 
 ```k
-    syntax Int     ::= #get (ByteMap , Int     ) [function, functional, smtlib(mapGet)]
-    syntax ByteMap ::= #set (ByteMap , Int, Int) [function, functional, smtlib(mapSet)]
- // -----------------------------------------------------------------------------------
-    rule #get( ByteMap <| M |>, KEY ) => {M [KEY]}:>Int requires         KEY in_keys(M) [concrete]
-    rule #get( ByteMap <| M |>, KEY ) => 0              requires notBool KEY in_keys(M)
+    syntax Int     ::= #get ( ByteMap , Int       ) [function, functional, smtlib(mapGet)]
+    syntax ByteMap ::= #set ( ByteMap , Int , Int ) [function, functional, smtlib(mapSet)]
+ // --------------------------------------------------------------------------------------
+    rule #get( ByteMap <| M |> , KEY ) => {M [KEY]}:>Int requires KEY in_keys(M) [concrete]
+    rule #get( ByteMap <| M |> , KEY ) => 0              [owise]
```

Comment:
So if I understand correctly, the cast to `Int` is an implicit side condition. Could we make it an explicit side condition instead? 

    rule #get( ByteMap <| M |> , KEY ) => {M [KEY]}:>Int requires          KEY in_keys(M) andBool isInt(M[KEY])
    rule #get( ByteMap <| M |> , KEY ) => 0              requires notBool (KEY in_keys(M) andBool isInt(M[KEY]))

Reply:
We could, but not quite how you've done here.

Unfortunately, this change causes a 25% slowdown in the proving. So I'm adding back in the original rule, but also adding the `owise` rule. My thinking is that in cases where the prover can _show_ it's not in the map, it will take the faster original rule, and otherwise it will take the slower owise rule. So we make it actually total, but also keep the speedup (hopefully).